### PR TITLE
Init PeerManager

### PR DESCRIPTION
**Pull request by danielgranhao**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -166,6 +179,22 @@ impl LightningNode {
         let _graph = persister.read_graph();
 
         // Step 12. Initialize the PeerManager
+        let mut ephemeral_bytes = [0; 32];
+        OsRng.try_fill_bytes(&mut ephemeral_bytes).map_err(|e| {
+            InitializationError::SecretGeneration {
+                message: e.to_string(),
+            }
+        })?;
```
Comment (gcomte) [inline src/lib.rs:187]: Should we move out the crypto stuff into its own module where it can be reused?

File: src/lib.rs [hunk 2]
```diff
@@ -166,6 +179,22 @@ impl LightningNode {
         let _graph = persister.read_graph();
 
         // Step 12. Initialize the PeerManager
+        let mut ephemeral_bytes = [0; 32];
+        OsRng.try_fill_bytes(&mut ephemeral_bytes).map_err(|e| {
```
Comment (gcomte) [inline src/lib.rs:183]: Should we move out the crypto stuff into its own module where it can be reused?

**Review by gcomte — COMMENTED** [review]
Minor discussion point, LGTM otherwise

File: src/lib.rs (hunk 2 above)
Comment (danielgranhao) [reply src/lib.rs:183]: I was starting to do that and then realised that it wouldn't be that reusable due to the returned error. To make it reusable, we would still have to `map_err()` in both places this is used, and we would end up with almost the same repeated code.

Comment (andrei-21) [reply src/lib.rs:183]: I will have a look into it.

**Comment by danielgranhao** [comment]
LGTM

**Review by gcomte — APPROVED** [review]

**Pull request merged**